File: Leetcode/2948. Make Lexicographically Smallest Array by Swapping Elements.cpp
class Solution {
public:
    vector<int> lexicographicallySmallestArray(vector<int>& nums, int limit) {
        int n = nums.size();
        vector<pair<int, int>> mnums;
        for (int i = 0; i < n; i++) {
            mnums.push_back({nums[i], i});
        }
        sort(mnums.begin(), mnums.end());
        priority_queue<int, vector<int>, greater<int>> pq;  
        int i = 0, j;
        while (i < n) {
            j = i + 1;
            while (j < n && mnums[j].first - mnums[j - 1].first <= limit) {
                j++;
            }
            for (int l = i; l < j; l++) {
                pq.push(mnums[l].second);
            }
            for (int l = i; l < j; l++) {
                int idx = pq.top(); pq.pop();
                nums[idx] = mnums[l].first;
            }
            i = j;
        }
        return nums;
    }
};

File: Leetcode/2948. Make Lexicographically Smallest Array by Swapping Elements.py
class Solution:
    def lexicographicallySmallestArray(self, nums: List[int], limit: int) -> List[int]:
        n=len(nums)
        mnums=[]
        for i in range(n):
            mnums.append([nums[i],i])
        mnums.sort()
        i,j=0,0
        while i<n:
            j=i+1
            while j<n and mnums[j][0] - mnums[j-1][0]<=limit:
                j+=1
            idx=[]
            for l in range(i,j):
                idx.append(mnums[l][1])
            idx.sort()
            p=0
            for l in range(i,j):
                nums[idx[p]]=mnums[l][0]
                p+=1
            i=j
        return nums
        
